lib/events/athena/athena.go

```diff
@@ -0,0 +1,372 @@
+	QueryResultsS3 string
+	// Workgroup is Glue workgroup where Athena queries are executed (optional).
+	Workgroup string
+	// GetQueryResultsInterval is used to define how long query will wait before
+	// checking again for results status if previous status was not ready (optional).
+	GetQueryResultsInterval time.Duration
+	// LimiterRate defines rate at which search_event rate limiter is filled (optional).
+	LimiterRate float64
+	// LimiterBurst defines rate limit bucket capacity (optional).
+	LimiterBurst int
+
+	// Batcher settings.
+
+	// QueueURL is URL of SQS, which is set as subscriber to SNS topic (required).
+	QueueURL string
+	// BatchMaxItems defines how many items can be stored in single Parquet
+	// batch (optional).
+	// It's soft limit.
+	BatchMaxItems int
+	// BatchMaxInterval defined interval at which parquet files will be created (optional).
+	BatchMaxInterval time.Duration
+
+	// Clock is a clock interface, used in tests.
+	Clock clockwork.Clock
+	// UIDGenerator is unique ID generator.
+	UIDGenerator utils.UID
+
+	// TODO(tobiaszheller): add FIPS config in later phase.
+}
+
+// CheckAndSetDefaults is a helper returns an error if the supplied configuration
+// is not enough to setup Athena based audit log.
+func (cfg *Config) CheckAndSetDefaults() error {
+	// AWS restrictions (https://docs.aws.amazon.com/athena/latest/ug/tables-databases-columns-names.html)
+	const glueNameMaxLen = 255
+	if cfg.Database == "" {
+		return trace.BadParameter("Database is not specified")
+	}
+	if len(cfg.Database) > glueNameMaxLen {
+		return trace.BadParameter("Database name too long")
+	}
+	if !isAlphanumericOrUnderscore(cfg.Database) {
+		return trace.BadParameter("Database name can contains only alphanumeric or underscore characters")
+	}
+
+	if cfg.TableName == "" {
+		return trace.BadParameter("TableName is not specified")
+	}
+	if len(cfg.TableName) > glueNameMaxLen {
+		return trace.BadParameter("TableName too long")
+	}
+	// TableName is appended directly to athena query. That's why we put extra care
+	// that no weird chars are passed here.
+	if !isAlphanumericOrUnderscore(cfg.TableName) {
+		return trace.BadParameter("TableName can contains only alphanumeric or underscore characters")
+	}
+
+	if cfg.TopicARN == "" {
+		return trace.BadParameter("TopicARN is not specified")
+	}
+
+	if cfg.LocationS3 == "" {
+		return trace.BadParameter("LocationS3 is not specified")
+	}
+	if scheme, ok := isValidUrlWithScheme(cfg.LocationS3); !ok || scheme != "s3" {
+		return trace.BadParameter("LocationS3 must be valid url and start with s3")
+	}
+
+	if cfg.LargeEventsS3 == "" {
+		return trace.BadParameter("LargeEventsS3 is not specified")
+	}
+	if scheme, ok := isValidUrlWithScheme(cfg.LargeEventsS3); !ok || scheme != "s3" {
+		return trace.BadParameter("LargeEventsS3 must be valid url and start with s3")
+	}
+
+	if cfg.QueueURL == "" {
+		return trace.BadParameter("QueueURL is not specified")
```

Comment:
A pretty minor UX critique, but do we want to use lower-case names in these errors to match what `SetFromURL()` expects? `url.Values.Get()` is case sensitive so if a user tries to set e.g. `QueueURL` as written in the error message, it'll fail.

I don't know that we care to do anything about it especially since it would tie this to the semantics of `SetFromURL()` which isn't necessarily called, but figured it was worth a mention.

Reply:
I will leave it as it is. In first iteration we want to focus on making it deployable on Cloud. In next iterations we will work on making sure it's work on self-hosted, and in that case it's probably worth reworking approach with URL and use different one.